### PR TITLE
resolves issue #8 support generation based on interface

### PR DESCRIPTION
**Pull request by jasonman107**
resolves issue #8 support generation based on interface
- regrouped the DeltaBuilder.vm file, took the common code into DeltaBuilderBody.vm, and kept the same file for concrete class but created another one InterfaceDeltaBuilder.vm for Interfaces
- added a few classes to support code gen for interface
- added sample and junit test for it - ClientOrder and ClientOrderTest

### CODE REVIEW

File: core/src/main/java/org/namstorm/deltaforce/core/DeltaBuilder.java [hunk 1]
```diff
@@ -33,6 +33,17 @@
...
     /**
+     * same as apply() but also clear the deltas afterwards
+     * @return object that's been applied to
+     */
+    T applyAndClearDeltas();
+
+    /**
+     * clears out the deltas
+     */
+    void rollback();
```
Comment (dopsun) [inline core/src/main/java/org/namstorm/deltaforce/core/DeltaBuilder.java]: Rename this as clear? In previous method, applyAndClearDeltas(), the clear word is used.

File: processors/src/main/java/org/namstorm/deltaforce/annotations/processors/DeltaBuilderProcessor.java [hunk 2]
```diff
@@ -43,6 +44,14 @@
 public class DeltaBuilderProcessor
         extends AbstractProcessor {
 
+    private static final String IMPL = "Impl";
+
+    private static final Pattern GETTER_PATTERN = Pattern.compile("^get([a-zA-Z]+)$");
```
Comment (dopsun) [inline processors/src/main/java/org/namstorm/deltaforce/annotations/processors/DeltaBuilderProcessor.java]: These 3 patterns, may need to cover numbers. E.g. `getMethod1()`

File: core/src/main/java/org/namstorm/deltaforce/core/DeltaBuilder.java (hunk 1 above)
Comment (jasonman107) [reply core/src/main/java/org/namstorm/deltaforce/core/DeltaBuilder.java]: renamed to clearDeltas()

File: processors/src/main/java/org/namstorm/deltaforce/annotations/processors/DeltaBuilderProcessor.java (hunk 2 above)
Comment (jasonman107) [reply processors/src/main/java/org/namstorm/deltaforce/annotations/processors/DeltaBuilderProcessor.java]: good point, changed to include numbers as well

**Review by dopsun — APPROVED** [review]

**Pull request merged**